### DISCUSSION TITLE
retrive backup share

### DISCUSSION THREAD

**Discussion by Uponn** [Web3Auth/Web3Auth#472] [Help]
Is there a way to retrive the backup share programmatically using the sdk, when a user enables the MFA?

**Comment by himanshuchawla009** [comment]
we return a backup share for custom verifiers back to the application (dappShare) which you can store as a backup, plz go through this answer for more information about dappShare : https://github.com/orgs/Web3Auth/discussions/427#discussioncomment-3442052

**Reply by Uponn** [reply]
Great post, I stumbled upon it while researching but still didnt fully helped me, so how do I go on implementing this for the Web version?I've created a Custom Verifier and added it as an Adapter but still in the response I get an empty dappShare.

**Reply by himanshuchawla009** [reply]
what's your getUserInfo function response looks like?

**Reply by Uponn** [reply]
```
{
    "email": "uponnss@gmail.com",
    "name": "uponnss@gmail.com",
    "profileImage": "",
    "aggregateVerifier": "tkey-auth0-email-passwordless-lrc",
    "verifier": "torus",
    "verifierId": "uponnss@gmail.com",
    "typeOfLogin": "jwt",
    "dappShare": "",
    "idToken": "eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6IlRZT2dnXy01RU9FYmxhWS1WVlJZcVZhREFncHRuZktWNDUzNU1aUEMwdzAifQ.eyJpYXQiOjE2NjE0MTkxNzIsImF1ZCI6IkJPY21Nd09MdWoyN3lsMnlFczlVU19OR2ltdTg5eXJJaHdfV240ejlvVkE1STdsMmZmUTNLc3dnVW9mUnVNcUxvOGtoWkVJLWRyeDRpWDc5cGlfVkNvWSIsIm5vbmNlIjoiMDNkMzAyMTQ2NmFmOTM5NmMzM2VkODAxNmMzZWU0YTIyNDE5NGU4ZTRlYTc5YmE5OGNkOGRkMGQ3MmU2NjAwOGQ1IiwiaXNzIjoiaHR0cHM6Ly9hcGkub3BlbmxvZ2luLmNvbSIsIndhbGxldHMiOlt7InB1YmxpY19rZXkiOiIwMjg2NGYzNGU0ODJhOGQ1ZTZlYjZmMDNlYzMyYTJmYjRkZWE1ZThiMTE4OWI5MWZmODgwNWE0MThjNjg4ODZmNjYiLCJ0eXBlIjoid2ViM2F1dGhfYXBwX2tleSIsImN1cnZlIjoic2VjcDI1NmsxIn1dLCJlbWFpbCI6InVwb25uc3NAZ21haWwuY29tIiwibmFtZSI6InVwb25uc3NAZ21haWwuY29tIiwidmVyaWZpZXIiOiJ0b3J1cyIsInZlcmlmaWVySWQiOiJ1cG9ubnNzQGdtYWlsLmNvbSIsImFnZ3JlZ2F0ZVZlcmlmaWVyIjoidGtleS1hdXRoMC1lbWFpbC1wYXNzd29yZGxlc3MtbHJjIiwiZXhwIjoxNjYxNTA1NTcyfQ.oD8BJ-q5E8RV79j28_qfvHA6JEEc22CLgBfmsBs-ty-67JELAw-SmnY3FPUn-sM-BCdJXsVedPboy6jr-Kr11w",
    "oAuthIdToken": "",
    "oAuthAccessToken": ""
}
```

**Reply by Uponn** [reply]
also I keep on getting this error when I click on the Login button and select my gmail account
```
Could not get result from torus nodes Error occurred while verifying paramsazip is not clientID 308188786588-8g0gsi29dqsolquiaullgou4k0vmjkcd.apps.googleusercontent.com 308188786588-enf0fhbvrldvft8ckcmh94af59skn6ra.apps.googleusercontent.com
```

**Reply by himanshuchawla009** [reply]
your user info response shows that custom verifier is not being used  as verifier name is "torus" which is default verifier by torus, that should show your verifier name. 

I think you are not passing config correctly in web3auth sdk.

for the error, refer to this doc:- 

https://web3auth.io/docs/troubleshooting/jwt-errors#error-occurred-while-verifying-params-could-not-verify-identity

**Reply by Uponn** [reply]
Okay, I fixed the custom verifier, now I'm using my own
```
{
    "client": "BOcmMwOLuj27yl2yEs9US_NGimu89yrIhw_Wn4z9oVA5I7l2ffQ3KswgUofRuMqLo8khZEI-drx4iX79pi_VCoY",
    "currentLoginProvider": "google",
    "popupWindow": "false",
    "pid": "39125f23b2666170ae7617fabaa339f3fd0001dd7f800c2d4a6a23d48f203869",
    "whiteLabel": "{}",
    "keyMode": "v1",
    "isCustomVerifier": "true",
    "instanceId": "tj4ix4i2ag",
    "verifier": "weichain-google-verifier",
    "typeOfLogin": "google",
    "redirectToOpener": false
}
```
but the error persist, so my custom verifier Login Provider is set to Google I'm not using Custom so I do not have the JWT Verifier ID option

**Reply by Uponn** [reply]
@himanshuchawla009

**Reply by himanshuchawla009** [reply]
plz file an issue here with your code snippet (where you are passing verifier config) , verifier details and jwt token here: 
https://github.com/Web3Auth/Web3Auth

**Reply by Uponn** [reply]
@himanshuchawla009 hey, so its working now, thanks, but could you explain to me how does dappShare work, cuz I see that when I activate the 2FA from my app I don't see it activated when I visit https://app.openlogin.com/ and login with my account that I just activated 2FA with, also the 24 phrases are different from the ones that I was send on the email and the first couple of times I logout and login they are different and then they stop changing?

**Reply by chaitanyapotti** [reply]
dappShare can only be returned after the user sets up 2fa for the account. 
the accounts on https://app.openlogin.com don't belong to your custom verifiers. hence, you see different phrases etc.
